ecs-agent/wsclient/client_factory.go

```diff
@@ -0,0 +1,13 @@
+package wsclient
+
+import (
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+)
+
+// ClientFactory interface abstracts the method that creates new ClientServer
+// objects. This is helpful when writing unit tests.
+type ClientFactory interface {
+	New(url string, credentialProvider *credentials.Credentials, rwTimeout time.Duration) ClientServer
```

Comment:
This needs to be updated to include a `*wsclient.WSClientMinAgentConfig` parameter in the function signature.